### PR TITLE
copy-dbacredential: add support for cryptographic providers

### LINKED ISSUE #7896: Enhance copy-dbacredential for cryptographic providers
### Summarize Functionality

I am trying to copy a credential. The copy-dbacredential works fine except the credential I am trying to copy was created with the FOR CRYPTOGRAPHIC PROVIDER - the cyyptographic provider info for the credential does not copy over to the new server. I've looked at the code (ps1) for copy-dbacredential and of course it does not include the FOR CRYPTOGRAPHIC PROVIDER. Can the copy-dbacredential function be enhanced to copy the cryptographic provider across to a new SQL Server instance if there is one specified please

### Is there a command that is similiar or close to what you are looking for?

Yes

### Technical Details

For example:

CREATE CREDENTIAL AlterEgo WITH IDENTITY = 'Mary5',   
    SECRET = '<EnterStrongPasswordHere>'
FOR CRYPTOGRAPHIC PROVIDER "AzureEKM"

### PR DESCRIPTION
**Pull request by lancasteradam**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes # )
 - [x] New feature (non-breaking change, adds functionality, fixes #7896 )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### Purpose

Enable the copy of credentials that use a crypto provider as requested in #7896 

### Approach

Minor change to check if a crypto provider was used in the source credential and then apply that to the destination/copied credential. If the destination instance does not have the crypto provider than a warning is returned.

### Commands to test

See the tests for details. There is a conditional based on whether the test environment has the crypto providers configured at the instance level. It seems safer to assume that the crypto providers are installed separately from the pester test.

### CODE REVIEW

File: functions/Copy-DbaCredential.ps1 [hunk 1]
```diff
@@ -127,7 +127,6 @@ function Copy-DbaCredential {
             #>
             [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidUsingPlainTextForPassword", "", Justification = "For Credentials")]
             param (
-                [string[]]$Credentials,
```
Comment (lancasteradam) [inline functions/Copy-DbaCredential.ps1:130]: removed this as it seems unused

File: functions/Copy-DbaCredential.ps1 [hunk 2]
```diff
@@ -202,15 +212,6 @@ function Copy-DbaCredential {
         }
 
         Invoke-SmoCheck -SqlInstance $sourceServer
-
-        Write-Message -Level Verbose -Message "Checking if Remote Registry is enabled on $Source"
```
Comment (lancasteradam) [inline functions/Copy-DbaCredential.ps1:206]: removed this section as it seems unnecessary for the command functionality

File: functions/Copy-DbaCredential.ps1 [hunk 3]
```diff
@@ -223,7 +224,7 @@ function Copy-DbaCredential {
             }
             Invoke-SmoCheck -SqlInstance $destServer
 
-            Copy-Credential $credentials -force:$force
```
Comment (lancasteradam) [inline functions/Copy-DbaCredential.ps1:226]: removed this as it seems unused

File: tests/Copy-DbaCredential.Tests.ps1 [hunk 4]
```diff
@@ -5,30 +5,68 @@ Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
...
+            sp_configure 'EKM provider enabled', 1;
+            GO
+            RECONFIGURE;
+
+            2. Install https://www.microsoft.com/en-us/download/details.aspx?id=45344 on the instance2 and instance3 servers.
+
+            3. Run these SQL commands on the instance2 and instance3 servers:
+
+            CREATE CRYPTOGRAPHIC PROVIDER dbatoolsci_AKV
+            FROM FILE = 'C:\Program Files\SQL Server Connector for Microsoft Azure Key Vault\Microsoft.AzureKeyVaultService.EKM.dll';
```
Comment (potatoqualitee) [inline tests/Copy-DbaCredential.Tests.ps1:58]: This is brilliant and necessary for something I needed this week, thank you!

**Comment by potatoqualitee** [comment]
Looks good, thank you

**Pull request merged**